Improve next step in run log modal [PREMIUM-280]

NeosinneR opened the pull request:
## Description

See https://github.com/mailpoet/mailpoet-premium/pull/897

## Code review notes

_N/A_

## QA notes

See https://github.com/mailpoet/mailpoet-premium/pull/897

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/897

## Linked tickets

[PREMIUM-280]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] 🚫 I added sufficient test coverage
- [x] 🚫 I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[PREMIUM-280]: https://mailpoet.atlassian.net/browse/PREMIUM-280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

websupporter reviewed (dismissed):
lgtm. thank you @NeosinneR

NeosinneR committed:
Properly close activity modal when there is an error in API

[PREMIUM-280]

NeosinneR committed:
Show link to Scheduled Actions when the automation run is past due when it was suppose to run

[PREMIUM-280]

NeosinneR committed:
Remove next step from activity modal

The previous assumption was, that a Delay action is not in mailpoet_automation_run_log database, and had to be calculated. But this was a wrong assumption, and since the action is there, we don't have to calculate it.

[PREMIUM-280]

NeosinneR committed:
Show time left for step in activity modal

[PREMIUM-280]

NeosinneR committed:
Update sample run logs data structure

[PREMIUM-280]

websupporter reviewed (approved):
lgtm. thank you @NeosinneR

Aschepikov reviewed (approved):
GJ!
